src/main/java/gregtech/api/items/toolitem/ItemGTToolbelt.java

```diff
@@ -0,0 +1,768 @@
+            else return null;
+        }
+
+        @Override
+        public NBTTagCompound serializeNBT() {
+            return this.getHandler(0).serializeNBT();
+        }
+
+        @Override
+        public void deserializeNBT(NBTTagCompound nbt) {
+            // make sure we can load all the slots, no matter what we're supposed to be limited to.
+            int minsize = nbt.hasKey("Size") ? nbt.getInteger("Size") : 0;
+            // .copy() prevents double damage ticks in singleplayer
+            this.getHandler(minsize).deserializeNBT(nbt.copy());
+        }
+
+        protected ToolStackHandler getHandler(int minsize) {
+            int slots = Math.max(slotCountSupplier.get(), minsize);
+            if (handler == null || handler.getSlots() != slots) handler = new ToolStackHandler(slots);
+            return handler;
+        }
+    }
+
+    @Override
+    public NBTTagCompound getNBTShareTag(ItemStack stack) {
+        NBTTagCompound tag = new NBTTagCompound();
+        if (stack.getTagCompound() != null) {
+            tag.setTag("NBT", stack.getTagCompound());
+        }
+        tag.setTag("Cap", getHandler(stack).serializeNBT());
+        return tag;
+    }
+
+    @Override
+    public void readNBTShareTag(ItemStack stack, NBTTagCompound nbt) {
+        // cap syncing is handled separately, we only need it on the share tag so that changes are detected properly.
+        stack.setTagCompound(nbt == null ? null : (nbt.hasKey("NBT") ? nbt.getCompoundTag("NBT") : null));
+    }
+
+    protected static class ToolStackHandler extends ItemStackHandler {
+
+        private Ingredient nextCraftIngredient;
+
+        private static final Set<String> EMPTY = ImmutableSet.of();
+
+        private @Nullable Integer selectedSlot = null;
+
+        protected final ItemTool[] tools = new ItemTool[this.getSlots()];
+        protected final IGTTool[] gtTools = new IGTTool[this.getSlots()];
+        protected final Set<String> toolClasses = new ObjectOpenHashSet<>();
+        public final Set<String> oreDicts = new ObjectOpenHashSet<>();
+
+        private boolean passthrough = true;
+
+        public ToolStackHandler(int size) {
+            super(size);
+        }
+
+        public void incrementSelectedSlot() {
+            for (int slot = (this.selectedSlot == null ? -1 : this.selectedSlot) + 1; slot != this.getSlots(); slot++) {
+                if (this.getStackInSlot(slot).isEmpty()) continue;
+                this.selectedSlot = slot;
+                return;
+            }
+            this.selectedSlot = null;
+        }
+
+        public void decrementSelectedSlot() {
+            for (int slot = (this.selectedSlot == null ? this.getSlots() : this.selectedSlot) - 1; slot != -1; slot--) {
+                if (this.getStackInSlot(slot).isEmpty()) continue;
+                this.selectedSlot = slot;
+                return;
+            }
+            this.selectedSlot = null;
+        }
```

Comment:
are these for loops necessary? you could probably do something like `GhostCircuitSyncHandler`'s `getNextCircuitValue(int delta)`